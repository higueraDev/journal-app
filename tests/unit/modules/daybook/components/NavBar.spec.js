const { shallowMount } = require("@vue/test-utils");
import NavBar from "@/modules/daybook/components/Navbar.vue";
import createVuexStore from "../../../mock-data/mockStore.js";

const mockRouter = {
	push: jest.fn(),
};
jest.mock("vue-router", () => ({
	useRouter: () => mockRouter,
}));

describe("Pruebas en el NavbarComponent", () => {
	const store = createVuexStore({
		status: "authenticated",
		user: {
			name: "test",
			email: "test@test.com",
		},
		idToken: "ABC-123",
		refreshToken: "XYZ-123",
	});

	beforeEach(() => jest.clearAllMocks());

	test("Debe mostrar el componente correctamente", () => {
		const wrapper = shallowMount(NavBar, {
			global: {
				plugins: [store],
			},
		});

		expect(wrapper.html()).toMatchSnapshot();
	});

	test("Logout debe cerrar sesion y redireccionar", async () => {
		const wrapper = shallowMount(NavBar, {
			global: {
				plugins: [store],
			},
		});

		await wrapper.find("button").trigger("click");

		expect(mockRouter.push).toHaveBeenCalledWith({ name: "login" });
		expect(store.state.authModule).toEqual({
			status: "not-authenticated",
			user: null,
			idToken: null,
			refreshToken: null,
		});
	});
});
