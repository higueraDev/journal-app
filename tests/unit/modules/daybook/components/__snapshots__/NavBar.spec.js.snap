// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el NavbarComponent Debe mostrar el componente correctamente 1`] = `
<nav class="navbar bg-primary"><a class="navbar-brand text-white"><img src="" class="d-inline-block align-text-top mx-2" alt="Vue Logo" height="24"> test</a>
  <div class="d-flex"><button class="btn btn-outline-info"><i class="fa fa-sign-out"></i></button></div>
</nav>
`;
